src/Workspaces/CSharp/Portable/Extensions/MemberDeclarationSyntaxExtensions.cs

```diff
@@ -331,7 +331,10 @@ public static ArrowExpressionClauseSyntax GetExpressionBody(this MemberDeclarati
             switch (memberDeclaration?.Kind())
             {
                 case SyntaxKind.PropertyDeclaration:
-                    return ((PropertyDeclarationSyntax)memberDeclaration).ExpressionBody;
+                    var propertyDeclaration = (PropertyDeclarationSyntax)memberDeclaration;
+                    return propertyDeclaration.ExpressionBody ??
+                        propertyDeclaration.AccessorList.Accessors.FirstOrDefault(a => a.ExpressionBody != null)
```

Comment:
@CyrusNajmabadi this could be rewritten to just use SemanticModel.GetEnclosingSymbol and check that it's a non-static non-type symbol. Thoughts?